pkg/kdump/kdump-view.jsx

```diff
@@ -201,78 +202,32 @@ export class KdumpPage extends React.Component {
         if (key == "compression") {
-            if (value) {
-                // enable compression
-                if ("core_collector" in settings)
-                    settings.core_collector.value = settings.core_collector.value + " -c";
-                else
-                    settings.core_collector = { value: "makedumpfile -c" };
-            } else {
-                // disable compression
-                if ("core_collector" in this.props.kdumpStatus.config) {
-                    // just remove all "-c" parameters
-                    settings.core_collector.value =
-                        settings.core_collector.value
-                                .split(" ")
-                                .filter((e) => { return (e != "-c") })
-                                .join(" ");
-                } else {
-                    // if we don't have anything on this in the original settings,
-                    // we can get rid of the entry altogether
-                    delete settings.core_collector;
-                }
-            }
+            settings.compression.enabled = value;
         } else if (key === "target") {
             /* target changed, restore settings and wipe all settings associated
              * with a target so no conflicting settings remain */
             settings = {};
+            // TODO: do we need a deep copy here?
             Object.keys(this.props.kdumpStatus.config).forEach((key) => {
                 settings[key] = { ...this.props.kdumpStatus.config[key] };
             });
-            Object.keys(this.props.kdumpStatus.target).forEach((key) => {
-                if (settings[key])
-                    delete settings[key];
-            });
-            if (value === "ssh")
-                settings.ssh = { value: "" };
-            else if (value === "nfs")
-                settings.nfs = { value: "" };
-
-            if ("core_collector" in settings &&
-                settings.core_collector.value.includes("makedumpfile")) {
-                /* ssh target needs a flattened vmcore for transport */
-                if (value === "ssh" && !settings.core_collector.value.includes("-F"))
-                    settings.core_collector.value += " -F";
-                else if (settings.core_collector.value.includes("-F"))
-                    settings.core_collector.value =
-                        settings.core_collector.value
-                                .split(" ")
-                                .filter(e => e != "-F")
-                                .join(" ");
-            }
+            settings.targets = {};
+            settings.targets[value] = { type: value };
         } else if (key !== undefined) {
+            const type = Object.keys(settings.targets)[0];
             if (!value) {
-                if (settings[key])
-                    delete settings[key];
+                if (settings.targets[type][key])
+                    delete settings.targets[type][key];
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17544-20220719-143653-7a0d5780-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)